File: 318. Maximum Product of Word Lengths.py
class Solution(object):
    def maxProduct(self, words):
        """
        :type words: List[str]
        :rtype: int
        """
        wordlist = []
        for i in words:
            letters = 0
            for j in i:
                letters|= 1 << ord(j) - ord('a')
            wordlist += [letters]
        res = 0
        for i in xrange(len(wordlist)):
            for j in xrange(i+1, len(wordlist)):
                if wordlist[i] & wordlist[j] == 0:
                    res = max(res, len(words[i]) * len(words[j]))
        return res

File: 318. Maximum Product of Word Lengths.cpp
class Solution {
public:
    int maxProduct(vector<string>& words) {
        vector <vector<int>> wordlist;
        int res = 0;
        for (int i = 0; i < words.size(); i++)
        {
            vector <int> letters (26);
            for (int j = 0; j < words[i].size(); j++)
                letters[words[i][j] - 'a'] = 1;
            wordlist.push_back(letters);
        }
        for (int i = 0; i < wordlist.size(); i++)
        {
            for (int j = i+1; j < wordlist.size(); j++)
            {
                int difference = 0;
                for (int k = 0; k < 26; k++)
                {
                    if (wordlist[i][k] == 1 && wordlist[j][k] == 1)
                        break;
                    else
                        difference ++;
                }
                if (difference == 26)
                    res = res > words[i].size() * words[j].size() ? res : words[i].size() * words[j].size();
            }
        }
        return res;
    }
};
